### PR TITLE
Fixes light eaters not consuming gunlights / helmet lights

### LINKED ISSUE #53152: Nightmares cannot destroy gun-mounted lights.
## Round ID:

 [144853](https://scrubby.melonmesa.com/round/144853)

## Reproduction:

 Be nightmare, attack man with gunlight on his gun. Die because you cannot destroy that light.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #53152

:cl: ShizCalev
fix: Fixed light eaters not consuming helmet / gun lights.
/:cl:

### CODE REVIEW

**Comment by Shirbu** [comment]
noice

File: code/modules/mob/living/carbon/human/species_types/shadowpeople.dm [hunk 1]
```diff
@@ -223,16 +220,29 @@
...
 
+///checks if the item has an active light, and destroy the light source if it does.
+/obj/item/light_eater/proc/light_item_check(obj/item/I, atom/A)
+	if(!isitem(I))
+		return
+	if(I.light_range && I.light_power)
+		disintegrate(I, A)
+	else if(istype(I, /obj/item/gun))
+		var/obj/item/gun/G = I
+		if(G.gun_light?.on)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/carbon/human/species_types/shadowpeople.dm:240]: why are you checking if on is null in an if block?

Comment (Rohesie) [reply code/modules/mob/living/carbon/human/species_types/shadowpeople.dm:240]: What do you mean? They're checking if the gun has a `gun_light`, and if so, if it's on.

Comment (LemonInTheDark) [reply code/modules/mob/living/carbon/human/species_types/shadowpeople.dm:240]: Ah I see, I got twisted on the syntax

**Pull request merged**